include/boost/fusion/support/unused.hpp

```diff
@@ -34,7 +34,7 @@ namespace boost { namespace fusion
         }
 
         template <typename T>
-        BOOST_CONSTEXPR BOOST_FUSION_GPU_ENABLED
+        BOOST_FUSION_CONSTEXPR_THIS BOOST_FUSION_GPU_ENABLED
```

Comment:
Yes, #11517 has two distinct issue: one is 'cannot combine `constexpr` and `const`' in variable decl., another is GCC doesn't accept use of 'this' in `constexpr`.